packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1029,8 +1033,49 @@ test.serial('writeGranulesFromMessage() stores error on granule if any file fail
     knex,
     { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
   );
-  t.is(pgGranule.error.Error, 'Failed writing files to PostgreSQL.');
-  t.true(pgGranule.error.Cause.includes('AggregateError'));
+  const pgGranuleError = JSON.parse(pgGranule.error.errors);
+  t.deepEqual(pgGranuleError.map((error) => error.Error), ['Failed writing files to PostgreSQL.']);
+  t.true(pgGranuleError[0].Cause.includes('AggregateError'));
+});
+
+test.serial('writeGranulesFromMessage() stores an aggregate workflow error and file-writing error on a granule', async (t) => {
+  const {
+    cumulusMessage,
+    knex,
+    collectionCumulusId,
+    executionCumulusId,
+    providerCumulusId,
+    granuleModel,
+    granuleId,
+  } = t.context;
+
+  cumulusMessage.meta.status = 'failed';
+  cumulusMessage.exception = { Error: 'Unknown error', Cause: { Error: 'Workflow failed' } };
+  cumulusMessage.payload.granules[0].files[0].bucket = undefined;
+  cumulusMessage.payload.granules[0].files[0].key = undefined;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const dynamoGranule = await granuleModel.get({ granuleId });
+  const dynamoGranuleError = JSON.parse(dynamoGranule.error.errors);
+  t.is(dynamoGranule.status, 'failed');
+  t.deepEqual(dynamoGranuleError.map((error) => error.Error), ['Unknown error', 'Failed writing files to PostgreSQL.']);
+  t.truthy(dynamoGranuleError.map((error) => error.Cause));
```

Comment:
`.map()` returns an array and even if there were no values for `error.Cause`, it would return `[]`. However, `[]` **is a truthy value in Javascript**:

https://developer.mozilla.org/en-US/docs/Glossary/Truthy

So I'm not sure what this assertion proves.

It might be more valuable to explicitly compare `dynamoGranuleError[0].Cause` against the expected value of `{ Error: 'Workflow failed' }`.  I'm not sure if the the second error has a `Cause` property that we could meaningfully compare against, but if not, then I don't think it is essential

Reply:
[`67e428e` (#2567)](https://github.com/nasa/cumulus/pull/2567/commits/67e428e8e51eb52c3892d1b41e37e635d32c23fe)